middle_end/flambda/simplify/simplify_named.rec.ml

```diff
@@ -181,6 +181,9 @@ let simplify_named0 dacc (bindable_let_bound : Bindable_let_bound.t)
     in
     let kind = P.result_kind' prim in
     let dacc =
+      (* CR mshinwell: It's a bit weird that the env_extension is added to
+         the typing env here; couldn't it just have been returned already
+         added to [dacc]? *)
```

Comment:
The only case I've found when the `dacc` returned is not the one passed as argument to `simplify_primitive` is in `simplify_project_var`, where the used closure vars are updated (although I haven't looked exhaustively). Maybe we could simply not return `dacc` from `simplify_*` functions ?

Reply:
Let's see how things develop, as discussed on Slack.